### PR TITLE
Add and remove

### PR DESCRIPTION
**Pull request by victormc13**
# Latest Features

### Project requirements
- Remove all hardcoded items from the tasks array.
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for editing task descriptions.
- By default new tasks should have the property `completed` set to `false` and the property `index` set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1)..
- All changes to the To Do List should be saved in local storage.

### CODE REVIEW

File: src/modules/tasks.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
+const tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+
+function saveTasks() {
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+
+export function renderTasks() {
```
Comment (Ismailco) [inline src/modules/tasks.js:7]: - [ ] Kindly make sure to break your code down, you should have three different functions for adding, editing and deleting tasks from your app.

File: src/modules/tasks.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    taskContainer.appendChild(taskItem);
+  });
+}
+
+export function addTask(description) {
+  const index = tasks.length;
+  tasks.push({
+    description,
+    completed: false,
+    index,
```
Comment (Ismailco) [inline src/modules/tasks.js:137]: -  [ ] The index should start from 1 not 0, kindly consider fixing this so the array starts with index 1, so if there are 5 tasks in the list the last task should have an array of 5 not 4

Screenshot_20230516_061908

File: src/modules/tasks.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      taskMenu.classList.add('hidden');
+
+      if (taskItem.classList.contains('task-editing')) {
+        trashIcon.classList.remove('hidden');
+        trashIcon.classList.add('show');
+
+        taskDescription.contentEditable = true;
+        taskDescription.focus();
+
+        trashIcon.addEventListener('click', () => {
```
Comment (Ismailco) [inline src/modules/tasks.js:78]: - [ ] Deleting one item from the list should re-arrange the whole list and reassign new indexes starting from 1, kindly make sure to fix this as for now the indexes are not re-arranging

Screenshot_20230516_062210

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @victormc13, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
- [ ] Kindly make sure to remove the Live demo section from your README if you're now planning on deploying the website now

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @victormc13 

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀

image


## Highlights 
- Proper use of GitHub PR  features ✔️ 
- Kudus on implementing  changes suggested by the last reviewer ✔️ 

## Confirmed index and re-arranging of the index after deletion:
image


## Optional Suggestions 💡
While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really enhance your application.
Cheers, and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread, and don't forget to tag me so I'll receive the notification.

As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**